packages/pyxis-react/src/components/Modal/Header/Header.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, ReactElement, ReactNode } from 'react';
+import Button from 'components/Button';
+import classNames from 'classnames';
+import { IconClose } from 'components/Icon/Icons';
+import { IconProps } from 'components/Icon';
+import { BadgeProps } from 'components/Badge';
+
+const getClasses = (isSticky: boolean): string => classNames('modal__header', { 'modal__header--sticky': isSticky });
+
+const getWrapperClasses = (isCustom: boolean) => classNames('modal__header__wrapper', {
+  'modal__header__wrapper--custom': isCustom,
+});
+export const Header: FC<HeaderProps> = ({
+  badge,
+  children,
+  icon,
+  id,
+  isSticky = false,
+  onClose,
+  title,
+}) => (
+  <header className={getClasses(isSticky)} id={id}>
+    <div
+      className={getWrapperClasses(!!children)}
+      data-testid={`${id}-wrapper`}
+    >
+      {children || (
+        <>
+          {badge && <div className="modal__header__badge">{badge}</div>}
+          {icon}
+          {title && <h3 className="modal__header__title">{title}</h3>}
+        </>
+      )}
+    </div>
+    {onClose && (
+      <Button
+        icon={IconClose}
+        iconPlacement="only"
+        onClick={onClose}
+        size="medium"
+        variant="tertiary"
+      >
+        Close
+      </Button>
+    )}
+  </header>
+);
+
+export default Header;
+
+export interface HeaderProps {
+  badge?: ReactElement<BadgeProps>;
+  icon?: ReactElement<IconProps>;
```

Comment:
Maybe is better define this with FC (like components `Button.tsx`)
```suggestion
  icon?: FC<IconProps>;
```

Reply:
As Cory explained [here](https://github.com/primait/pyxis/pull/177#discussion_r890212372), the ReactElement is necessary if we want the user to customize the Icon

Reply:
This works for `Button` as the `Icon` props are determined from other `Button` props.
https://github.com/primait/pyxis/blob/66f60710d1c35188a2de161fbbd2582eafa07207/packages/pyxis-react/src/components/Button/Button.tsx#L27-L30
Changing this to a `FC` would mean the `Icon` could only ever be its default state unless another `iconProps` prop was added to `Modal.Header`, at which point we are just achieving `ReactElement` but with more steps. The only benefit I can see from this would be better typing which I mentioned in my reply to Laura's feedback.
 
> Icon should be a `ReactElement` so that it can be configured externally. An example may be that you want to pass the `alt` prop to the `Icon` used, this can be done with `<Modal.Header ... icon={<IconCar alt/>}>`. Although `ReactElement` has some generic options for `props` and `type`, they don't seem to work. I've looked this up and it seems like a [known issue](https://stackoverflow.com/a/59381632).